src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error();
+    }
+
+    const condition = [this.a, this.b, this.c].sort((a1, a2) => a1 - a2);
+
+    if (condition[2] >= condition[0] + condition[1]) {
+      throw new Error();
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
 
+    return +Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c))
```

Comment:
Try to avoid using the unary operator to convert strings to numbers and try using something else instead